cassandra/src/main/java/se/tre/freki/storage/cassandra/CassandraStore.java

```diff
@@ -286,14 +286,24 @@ public void close() {
                                                final LabelId id,
                                                final LabelType type) {
     // Get old name
-    final ResultSetFuture oldNameFuture = session.executeAsync(
+    final ListenableFuture<List<Row>> oldNameFuture = getLabelRows(id, type);
 
-        getNameStatement.bind(toLong(id), type.toValue()));
-
-    return transform(oldNameFuture, new Function<ResultSet, LabelId>() {
+    return transform(oldNameFuture, new Function<List<Row>, LabelId>() {
       @Override
-      public LabelId apply(final ResultSet rows) {
-        final String oldName = rows.one().getString("name");
+      public LabelId apply(final List<Row> rows) {
+
+        //we only want to change the oldest label. It is the only label we care about
+        Date oldestDate = rows.get(0).getDate("timestamp");
+        String oldName = rows.get(0).getString("name");
+
+        for (Row row : rows) {
```

Comment:
All the checks in this for loop are redundant. The labels are explicitly ordered a certain way in the table and also returned in that order. This means that the name and date we get from `rows.get(0)` will always be the oldest one.